arm-sql/2014-04-01/swagger/capabilities.json

```diff
@@ -0,0 +1,332 @@
+                    "x-ms-enum": {
+                        "name": "CapabilityStatus"
+                    }
+                },
+                "supportedServiceLevelObjectives": {
+                    "readOnly": true,
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/ServiceObjectiveCapability"
+                    },
+                    "description": "The list of supported Service Objectives for the edition."
+                }
+            },
+            "description": "Represents the server edition capabilities."
+        },
+        "ServerVersionCapability": {
+            "properties": {
+                "name": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "The server version name."
+                },
+                "status": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "The status of the server version.",
+                    "enum": [
+                        "Visible",
+                        "Available",
+                        "Default"
+                    ],
+                    "x-ms-enum": {
+                        "name": "CapabilityStatus"
+                    }
+                },
+                "supportedEditions": {
+                    "readOnly": true,
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/EditionCapability"
+                    },
+                    "description": "The list of supported server editions."
+                }
+            },
+            "description": "Represents the server capabilities."
+        },
+        "LocationCapabilities": {
+            "properties": {
+                "name": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "The region name."
+                },
+                "status": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "The status for the region with respect to Azure SQL.",
+                    "enum": [
+                        "Visible",
+                        "Available",
+                        "Default"
+                    ],
+                    "x-ms-enum": {
+                        "name": "CapabilityStatus"
+                    }
+                },
+                "supportedServerVersions": {
+                    "readOnly": true,
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/ServerVersionCapability"
+                    },
+                    "description": "The list of supported server versions."
+                }
+            },
+            "description": "Represents the capabilities for a region."
+        },
+        "Resource": {
```

Comment:
@olydis hey Johannes, I tried it out, and the linter is failing.  

I moved SqlSubResource and Resource definitions to a Resource.json file, and change the reference to it like this:
          "$ref": "./resource.json#/definitions/SqlSubResource"

 Validate-spec continues to pass, but I get a linter error back: file: 'file:///e%3A/Azure/azure-rest-api-specs/arm-sql/2014-04-01/swagger/firewallRules.json'
severity: 'Error'
message: 'Error generating client model: Could not find file 'C'
at: '1,1'
source: 'lint'

I'm using linter 0.6.0.  Am I doing the reference correctly?